src/plugProjectNishimuraU/DangoMushi.cpp

```diff
@@ -1026,122 +759,35 @@ void Obj::getShadowParam(ShadowParam&)
  */
 bool Obj::needShadow()
 {
-	/*
-	stwu     r1, -0x10(r1)
-	mflr     r0
-	stw      r0, 0x14(r1)
-	stw      r31, 0xc(r1)
-	mr       r31, r3
-	bl       needShadow__Q24Game9EnemyBaseFv
-	clrlwi.  r0, r3, 0x18
-	beq      lbl_802FC9A4
-	li       r3, 1
-	b        lbl_802FC9D8
-
-lbl_802FC9A4:
-	lfs      f1, 0x2c8(r31)
-	lfs      f0, lbl_8051D3A8@sda21(r2)
-	fcmpo    cr0, f1, f0
-	ble      lbl_802FC9D4
-	mr       r3, r31
-	bl       getStateID__Q24Game9EnemyBaseFv
-	cmpwi    r3, 1
-	beq      lbl_802FC9CC
-	cmpwi    r3, 2
-	bne      lbl_802FC9D4
-
-lbl_802FC9CC:
-	li       r3, 1
-	b        lbl_802FC9D8
-
-lbl_802FC9D4:
-	li       r3, 0
-
-lbl_802FC9D8:
-	lwz      r0, 0x14(r1)
-	lwz      r31, 0xc(r1)
-	mtlr     r0
-	addi     r1, r1, 0x10
-	blr
-	*/
+	s32 state;
+	if (EnemyBase::needShadow()) {
+		return true;
+	}
+	if (mShadowScale > 0.0f && ((state = getStateID(), state == 1) || state == 2)) {
```

Comment:
Please simplify or split these if statements (into inline, or separate it in function) if possible, as it is a bit confusing